OrbitVulkanLayer/VulkanLayerController.h

```diff
@@ -0,0 +1,454 @@
+    return dispatch_table_.QueuePresentKHR(queue)(queue, present_info);
+  }
+
+  void OnCmdBeginDebugUtilsLabelEXT(VkCommandBuffer command_buffer,
+                                    const VkDebugUtilsLabelEXT* label_info) {
+    if (dispatch_table_.IsDebugUtilsExtensionSupported(command_buffer)) {
+      dispatch_table_.CmdBeginDebugUtilsLabelEXT(command_buffer)(command_buffer, label_info);
+    }
+
+    CHECK(label_info != nullptr);
+    submission_tracker_.MarkDebugMarkerBegin(command_buffer, label_info->pLabelName,
+                                             {
+                                                 .red = label_info->color[0],
+                                                 .green = label_info->color[1],
+                                                 .blue = label_info->color[2],
+                                                 .alpha = label_info->color[3],
+                                             });
+  }
+
+  void OnCmdEndDebugUtilsLabelEXT(VkCommandBuffer command_buffer) {
+    submission_tracker_.MarkDebugMarkerEnd(command_buffer);
+    if (dispatch_table_.IsDebugUtilsExtensionSupported(command_buffer)) {
+      dispatch_table_.CmdEndDebugUtilsLabelEXT(command_buffer)(command_buffer);
+    }
+  }
+
+  void OnCmdDebugMarkerBeginEXT(VkCommandBuffer command_buffer,
+                                const VkDebugMarkerMarkerInfoEXT* marker_info) {
+    if (dispatch_table_.IsDebugMarkerExtensionSupported(command_buffer)) {
+      dispatch_table_.CmdDebugMarkerBeginEXT(command_buffer)(command_buffer, marker_info);
+    }
+    CHECK(marker_info != nullptr);
+    submission_tracker_.MarkDebugMarkerBegin(command_buffer, marker_info->pMarkerName,
+                                             {
+                                                 .red = marker_info->color[0],
+                                                 .green = marker_info->color[1],
+                                                 .blue = marker_info->color[2],
+                                                 .alpha = marker_info->color[3],
+                                             });
+  }
+
+  void OnCmdDebugMarkerEndEXT(VkCommandBuffer command_buffer) {
+    submission_tracker_.MarkDebugMarkerEnd(command_buffer);
+    if (dispatch_table_.IsDebugMarkerExtensionSupported(command_buffer)) {
+      dispatch_table_.CmdDebugMarkerEndEXT(command_buffer)(command_buffer);
+    }
+  }
+
+  // ----------------------------------------------------------------------------
+  // Layer enumeration functions
+  // ----------------------------------------------------------------------------
+
+  [[nodiscard]] VkResult OnEnumerateInstanceLayerProperties(uint32_t* property_count,
+                                                            VkLayerProperties* properties) {
+    // Vulkan spec dictates that we are only supposed to enumerate ourselve.
+    if (property_count != nullptr) {
+      *property_count = 1;
+    }
```

Comment:
So we only have one property? Could we have more? If that was the case, how could more than one property be set in `properties`.

Reply:
Ah I guess that `VkLayerProperties*` is an array?

Reply:
Exactly.